server/src/main/java/timely/adapter/accumulo/MetricAdapter.java

```diff
@@ -0,0 +1,95 @@
+package timely.adapter.accumulo;
+
+import org.apache.accumulo.core.client.lexicoder.LongLexicoder;
+import org.apache.accumulo.core.client.lexicoder.PairLexicoder;
+import org.apache.accumulo.core.client.lexicoder.StringLexicoder;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Mutation;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.security.ColumnVisibility;
+import org.apache.accumulo.core.util.ComparablePair;
+import timely.model.Metric;
+import timely.model.Tag;
+import timely.model.parse.TagListParser;
+import timely.model.parse.TagParser;
+
+import java.nio.ByteBuffer;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * Creates Accumulo data structures from {@link timely.model.Metric}
+ */
+public class MetricAdapter {
+
+    private static final PairLexicoder<String, Long> rowCoder = new PairLexicoder<>(new StringLexicoder(),
+            new LongLexicoder());
+
+    private static final TagParser tagParser = new TagParser();
+    private static final TagListParser tagListParser = new TagListParser();
+
+    public static final ColumnVisibility EMPTY_VISIBILITY = new ColumnVisibility();
+    public static final String VISIBILITY_TAG = "viz";
+
+    public static Mutation toMutation(Metric metric) {
+        final Mutation mutation = new Mutation(encodeRowKey(metric));
+
+        List<Tag> tags = metric.getTags();
+        Collections.sort(tags); // TODO check that this works
```

Comment:
This was really to fix a problem with a test case. The test checks that two JSON strings are equal and one string is hardcoded in the test. Since the order of the array isn't guaranteed anywhere, the test failed because it got serialized in a different order than when Jackson serialized the object. I think the real fix is to have good tests for equals and compareTo in the model objects and tests for JSON serialization formats, then just rely on .equals in all the other tests.